File: week#6/DetailsPrinter.java
// take input of name, age and create & write the details into the file 
import java.io.File;  // the File class
import java.io.IOException; // the IOException class to handle errors
import java.io.FileWriter; // class to write in a file
import java.util.Scanner; // class for input
public class DetailsPrinter {
    private String name;
    private int age;

    // getter for name
    String getName(Scanner input) {
        System.out.println("What is your name?");
        name = input.nextLine();
        return name;
    }
    // getter for age
    int getAge(Scanner input) {
        System.out.println("What is your age?");
        age = input.nextInt();
        return age;
    }

    // create a file
    void makeFile() {
    try {
        File read = new File("details.txt");
        if(read.createNewFile()) {
            System.out.println(read.getName() + " is created");
        }
        else {
            System.out.println("File already exists");
        }

    }
    catch (IOException a) {
        System.out.println("An error occured");
        a.printStackTrace();
    }
    }

    // write in a file:
    void writeFile() {
    try {
    FileWriter writer = new FileWriter("details.txt" , true);
        writer.write("Name," + name + "\nAge," + age);
        writer.close();
        System.out.println("Added Text Successful.");
    } catch (Exception e) {
        System.out.println("An Error Occured");
        e.printStackTrace();
    }
    }


    // read in a file (line-by-line)
    void ReadFile() {

        try {
        // creates file object
        File thefile = new File("details.txt");
        // Scanner object for reading a line
        Scanner fileRead = new Scanner(thefile);
        // loop to read all lines in a loop
        while(fileRead.hasNextLine()) {
            String data = fileRead.nextLine();
            System.out.println(data);
        }
        // closing to prevent unexpected behaviour
        fileRead.close();
        }

        catch (IOException o) {
            System.out.println("Error Occured.");
            System.out.println(o);

        }
    }



    // main method
    public static void main(String[] args) {
    DetailsPrinter user = new DetailsPrinter();
    // Scanner object
    Scanner input = new Scanner(System.in);
    // Scanner object for file_read

    user.getName(input);
    user.getAge(input);
    input.close();
    user.makeFile();
    user.writeFile();
    user.ReadFile();
    }

}

File: week#6/DetailsPrinter.cpp
// for input/output
#include<iostream>
// for file I/O handling
#include<fstream>
// namespace for simple string, input, output
using namespace std;

// main class
class DetailsPrinter {
// private modifier
private:
    string name;
    int age;
// public modifier
public:

    // name getter
    string getName() {
        cout<<"Enter your name: ";
        cin>>name;
        return name;
    }
    // age getter
    int getAge() {
        cout<<"Enter your age: ";
        cin>>age;
        return age;
    }

    // write in file method
    void writeFile() {
        // creating filewriter class
        ofstream fileWriter("details.txt");
        // writing in file
        fileWriter << "Name, "<< name << endl;
        fileWriter << "Age, " << age << endl;
        // closing file writing
        fileWriter.close();
    }

    void readFile() {
        // creating object
        ifstream fileRead("details.txt");
        // variable to store each line in file
        string i;
        // loop to print each line in a file
            while (getline(fileRead, i)) 
            {
                cout<<i<<endl;
            }
            // closing the fileRead
                fileRead.close();
        
    
    }

};

int main() {
    // creating an object of the class to use its functions
    DetailsPrinter file;
// getting name, age, 
    file.getName();
    file.getAge();
    
// writing in file
    file.writeFile();

// printing in the console what is in file
    file.readFile();

}